### PR TITLE
Test clean up - moving into a scenario-like approach

### PR DESCRIPTION
**Pull request by Scooletz**
Common methods for different scenarios moved to the base test class. An additional generic test class added for removing the clutter of double generic parameters. Now they can be declared on the class level, which is the most common scenario.

### CODE REVIEW

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_updating_a_saga_with_the_same_unique_property_value.cs [hunk 1]
```diff
@@ -16,26 +16,9 @@ public class When_updating_a_saga_with_the_same_unique_property_value : SagaPers
...
-            {
-                SetActiveSagaInstance(updatingContext, new SagaWithCorrelationProperty(), saga1);
-                saga1 = await persister.Get<SagaWithCorrelationPropertyData>(nameof(SagaWithCorrelationPropertyData.CorrelatedProperty), correlationPropertyData, updateSession, updatingContext);
-                SetActiveSagaInstance(updatingContext, new SagaWithCorrelationProperty(), saga1);
+            await SaveSaga(saga1);
 
-                await persister.Update(saga1, updateSession, updatingContext);
-                await updateSession.CompleteAsync();
-            }
+            await GetByCorrelationPropertyAndUpdate(nameof(SagaWithCorrelationPropertyData.CorrelatedProperty), correlationPropertyData, _ => { });
```
Comment (danielmarbach) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_updating_a_saga_with_the_same_unique_property_value.cs]: Should we update something here?

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs [hunk 2]
```diff
@@ -6,6 +6,19 @@
...
+        protected Task<TSagaData> GetByIdAndUpdate(Guid sagaId, Action<TSagaData> update, params Type[] availableTypes) => GetByIdAndUpdate<TSaga, TSagaData>(sagaId, update, availableTypes);
+        protected Task<TSagaData> GetByCorrelationPropertyAndUpdate(string correlatedPropertyName, object correlationPropertyData, Action<TSagaData> update) => GetByCorrelationPropertyAndUpdate<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData, update);
+        protected Task<TSagaData> GetByCorrelationPropertyAndComplete(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationPropertyAndComplete<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetByCorrelationProperty(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationProperty<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetById(Guid sagaId, params Type[] availableTypes) => GetById<TSaga, TSagaData>(sagaId, availableTypes);
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs]: `GetById` as well to follow the convention?

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_updating_a_saga_with_the_same_unique_property_value.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/When_updating_a_saga_with_the_same_unique_property_value.cs]: The original test didn't, but it makes sense to.

File: src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs (hunk 2 above)
Comment (danielmarbach) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs]: @SeanFeldman not sure I understand. Can you elaborate?

Comment (SeanFeldman) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs]: The "outer" `GetById` is forwarding to the "inner" `GetById`. We're getting by ID, so why not to call the outer method `GetById` as well.

Comment (danielmarbach) [inline src/NServiceBus.Persistence.ServiceFabric.Tests/ComponentTests/Sagas/SagaPersisterTests.cs]: I'm probably blind 
image

**Comment by danielmarbach** [comment]
Pulling this in when green. We can address https://github.com/Particular/NServiceBus.Persistence.ServiceFabric/pull/14#discussion_r107600977 in a dedicated PR

**Pull request merged**